DOC-12749 New REST API settings in Data Service (#3838)
couchbase/docs-server: modules/learn/pages/buckets-memory-and-storage/memory.adoc

Warmup behavior controls how and when data is loaded from disk into memory after a bucket or node restarts. The `warmupBehavior` setting determines this process:

* **background** (default): The bucket starts serving requests as soon as metadata and some data are loaded into memory, while the rest of the data continues loading in the background.
This allows for faster bucket availability and minimal downtime.
* **blocking**: The bucket does not serve any requests until all data is fully loaded into memory.
This ensures that all items are available immediately when the bucket becomes active, but can increase startup time.
* **none**: The warmup process is skipped, and the bucket serves only items that are already in memory.
Data is loaded on-demand as requests are made for items not yet loaded.

Choosing the appropriate warmup behavior lets you balance between quicker bucket availability and having more data pre-loaded for immediate access after a restart.

On a given Data Service node, a bucket's data is loaded in accordance with determinations made by the _access scanner_.

The access scanner periodically identifies the most frequently used keys and writes them to an access log.

> During server warmup, Server uses this log to prioritize loading the most-accessed documents into memory.

Reviewer: ```suggestion
During server warmup, Couchbase Server uses this log to prioritize loading the most-accessed documents into memory.
```

Instead, it deletes any existing access log, and exits.
If and when data-loading subsequently occurs, since no access log exists, loading occurs with no priority-order (this being, in cases of extremely high resident ratio, the more performative loading procedure).

> NOTE: The settings `warmup_min_items_threshold`, `warmup_min_memory_threshold`, and enabling access scanner are no more supported through `cbepctl`.

Reviewer: ```suggestion
NOTE: The settings `warmup_min_items_threshold`, `warmup_min_memory_threshold`, and enabling access scanner are no longer supported through `cbepctl`.
```

For more information about buckets and bucket types, see xref:buckets-memory-and-storage/buckets.adoc[Buckets].

> NOTE: The settings `mem_high_wat`, and `mem_low_wat` are no more supported through `cbepctl`.

Reviewer: ```suggestion
NOTE: The settings `mem_high_wat`, and `mem_low_wat` are no longer supported through `cbepctl`.
```

When sufficient memory is again available, data-ingestion resumes.

Items are selected for ejection based on metadata that each contains, indicating whether the item can be classified as _Not Recently Used_ (NRU).
If an item has not been recently used, it is a candidate for ejection.

The relationship of `memoryLowWatermark` and `memoryHighWatermark` to the bucket's overall memory quota is illustrated as follows:

[#tunable_memory]
image::buckets-memory-and-storage/tunableMemory.png[,416]

The default setting for `memoryLowWatermark` is 75%.
The default setting for `memoryHighWatermark` is 85%.
The default settings can be changed using the REST API.
See xref:rest-api:rest-bucket-create.adoc#memorylowwatermark[memoryLowWatermark] and xref:rest-api:rest-bucket-create.adoc#memoryhighwatermark[memoryHighWatermark].

[#expiry-pager]
== Expiry Pager

Scans for items that have expired, and erases them from memory and disk; after which, a _tombstone_ remains for a default period of 3 days.
The expiry pager runs every 10 minutes by default.

> NOTE: The setting `exp_pager_stime` is no more supported through `cbepctl`.

Reviewer: ```suggestion
The setting `exp_pager_stime` is no longer supported through `cbepctl`.
```
